docs/smart-contracts/getting-started/create-an-app.mdx

```diff
@@ -324,23 +325,30 @@ Current value: <strong id="current-value" aria-live="polite">???</strong><br />
 
 <script>
   import incrementor from "../contracts/soroban_increment_contract";
-
+  import { isAllowed, getUserInfo } from '@stellar/freighter-api';
   const button = document.querySelector("[data-increment]");
   const currentValue = document.querySelector("#current-value");
-
+  async function getPk() {
+    const { publicKey } = await getUserInfo();
+    return publicKey;
+  }
+  if (await isAllowed()) {
+    const publicKey = await getPk();
+    if (publicKey) incrementor.options.publicKey = publicKey;
+  }
```

Comment:
This is fine for now, but I think Freighter now has a `getPublicKey` or just `publicKey` method that we can use here to simplify this logic.

Reply:
You are correct, this code works: